gcloud/credentials.py

```diff
@@ -160,7 +170,66 @@ def _get_pem_key(credentials):
     return RSA.importKey(pem_text)
 
 
-def _get_signed_query_params(credentials, expiration, signature_string):
+def _get_signature_bytes(credentials, string_to_sign):
+    """Uses crypto attributes of credentials to sign a string/bytes.
+
+    :type credentials: :class:`client.SignedJwtAssertionCredentials`,
+                       :class:`service_account._ServiceAccountCredentials`,
+                       :class:`_GAECreds`
+    :param credentials: The credentials used for signing text (typically
+                        involves the creation of an RSA key).
+
+    :type string_to_sign: string
+    :param string_to_sign: The string to be signed by the credentials.
+
+    :rtype: bytes
+    :returns: Signed bytes produced by the credentials.
+    """
+    if _GAECreds is not None and isinstance(credentials, _GAECreds):
+        _, signed_bytes = app_identity.sign_blob(string_to_sign)
+        return signed_bytes
+    else:
```

Comment:
`isinstance(credentials, _GAECreds)` needs to make sense. If `_GAECreds` is not a subclass of `type`, it actually errors. We could avoid the error by falling back to `NoneType` instead of `None`, but it still wouldn't make sense.